azure-communication-ui/chat/src/main/java/com/azure/android/communication/ui/chat/redux/middleware/repository/MessageRepositoryMiddleware.kt

```diff
@@ -142,18 +146,41 @@ internal class MessageRepositoryMiddlewareImpl(
     private fun processParticipantsRemoved(
         action: ParticipantAction.ParticipantsRemoved,
         dispatch: Dispatch,
+        localParticipant: LocalParticipantInfoModel
     ) {
-        messageRepository.addMessage(
-            MessageInfoModel(
+        val isLocalParticipantRemoved = action.participants.any { removed ->
+            removed.userIdentifier.id == localParticipant.userIdentifier
+        }
+        if (isLocalParticipantRemoved) {
+            val localUserRemovedSystemMessage = MessageInfoModel(
                 internalId = System.currentTimeMillis().toString(),
-                participants = action.participants.map { it.displayName ?: "" },
+                participants = arrayListOf(localParticipant.displayName ?: "Local Participant"),
                 content = null,
                 createdOn = OffsetDateTime.now(),
                 senderDisplayName = null,
-                messageType = ChatMessageType.PARTICIPANT_REMOVED
+                messageType = ChatMessageType.PARTICIPANT_REMOVED,
+                isCurrentUser = true
             )
-        )
-        notifyUpdate(dispatch)
+            messageRepository.addMessage(localUserRemovedSystemMessage)
```

Comment:
LocalParticipantRemoved logic doesn't belong in MessageRepositoryMiddleware
It shouldn't need to fake a message (user should lose access to the thread, they wouldn't see it)

It probably should be in ChatServiceListener

Reply:
As we are following teams behaviour we are just making the user not being able to send any messages and not making them receive any message in the chat thread. But the user can scroll the previous messages.